examples/comp_w_dac.rs

```diff
@@ -59,7 +61,7 @@ fn main() -> ! {
     // * 0V   at p1 =>   0% duty
     // * VDDA at p1 => 100% duty
     loop {
-        dac.set_value(val);
+        dac.as_mut().set_value(val);
```

Comment:
Should we even go as far as implementing `Deref{Mut}` for `Observed`?

Reply:
Yep, `Deref` will be a really good QoL feature.